### ISSUE TITLE
how to load custom dataset?

### ISSUE THREAD

**Issue by smith-co** [s-mrb/siamese-mnist#1]
I am trying to adapt this example with a new dataset.

I have a directory where I have kept `anchor`, `positive`, and `negative` examples, i.e.

```
anchor/
1.jpg
2.jpg
3.jpg
....
```

```
positive/
1.jpg
2.jpg
3.jpg
....
```

```
negative/
1.jpg
2.jpg
3.jpg
....
```

How could I feed this dataset into this model? The given code snippet reads directly from `keras.datasets.mnist.load_data()`.

**Comment by s-mrb** [comment]
**How could I feed this dataset into this model?**

@smith-co this is Duplet Network and  there is no **anchor** in **Duplet Network**.
Perhaps a primer on Constrastive Losses might be helpful, read [this](https://archives.ismir.net/ismir2019/paper/000057.pdf) and see [this](https://media.springernature.com/lw685/springer-static/image/art%3A10.1007%2Fs11760-019-01612-3/MediaObjects/11760_2019_1612_Fig1_HTML.png) and compare it to [this](https://www.mathworks.com/help/examples/nnet/win64/TrainASiameseNetworkForDimensionalityReductionExample_03.png). 

**TL;DR**
> In short Contrastive Loss is a generic loss function which just contrasts the positive and negative labeled images, this contrast can be done via two variants (Duplet Loss and Triplet Loss). Duplet Loss is Contrastive Loss which make use of a pair of input images from Duplet Network (Two Networks each with its own input), Triplet Loss is Contrastive Loss which make use of input images from Triplet Network (Three Networks). 

**You want to adapt this to your needs**, I won't be doing ur homework but here r hints:
  - `tower_anchor` = `embedding_network(anchor)`
  - `tower_positive` = ...
  - `tower_negative` = ...

You should be ok from here.

Morever **if you want to load custom dataset**, Please have a look [here](https://keras.io/getting_started/intro_to_keras_for_engineers/#data-loading-amp-preprocessing). Everything should be clear.

**Comment by s-mrb** [comment]
tell me if
